examples/client-benchmark/main.cpp

```diff
@@ -188,63 +191,67 @@ int main(int argc, char **argv) try {
 	auto pc = createPeerConnection(config, ws, id);
 
 	// We are the offerer, so create a data channel to initiate the process
-	const string label = "benchmark";
-	cout << "Creating DataChannel with label \"" << label << "\"" << endl;
-	auto dc = pc->createDataChannel(label);
+	for (int i = 1; i <= params.dataChannelCount(); i++) {
+		const string label = "DC-" + std::to_string(i);
+		cout << "Creating DataChannel with label \"" << label << "\"" << endl;
+		auto dc = pc->createDataChannel(label);
+		receivedSizeMap.emplace(label, 0);
+		sentSizeMap.emplace(label, 0);
 
-	// Set Buffer Size
-	dc->setBufferedAmountLowThreshold(bufferSize);
+		// Set Buffer Size
+		dc->setBufferedAmountLowThreshold(bufferSize);
 
-	dc->onOpen([id, wdc = make_weak_ptr(dc)]() {
-		cout << "DataChannel from " << id << " open" << endl;
-		if (noSend)
-			return;
+		dc->onOpen([id, wdc = make_weak_ptr(dc), label]() {
+			cout << "DataChannel from " << id << " open" << endl;
+			if (noSend)
+				return;
 
-		if (enableThroughputSet)
-			return;
+			if (enableThroughputSet)
+				return;
 
-		if (auto dcLocked = wdc.lock()) {
-			try {
-				while (dcLocked->bufferedAmount() <= bufferSize) {
-					dcLocked->send(messageData);
-					sentSize += messageData.size();
+			if (auto dcLocked = wdc.lock()) {
+				try {
+					while (dcLocked->bufferedAmount() <= bufferSize) {
+						dcLocked->send(messageData);
+						sentSizeMap[label] += messageData.size();
```

Comment:
Operator [] is basically an insert, which is not thread-safe. Sure, here you took care of emplacing elements before, so it should work in practice, however I think it would still be cleaner to access the unordered map in a read-only with `at`.

Reply:
I am not sure If I got you correctly. Normally online one thread changes one map value, and this should be safe.
Anyway I changed the logic to;
`sentSizeMap[label] = sentSizeMap.at(label) + messageData.size();`

Reply:
My point is the operator [] is technically an insert, so it would be more elegant to use the read-only method `at()` to retrieve the reference on the atomic like this:
```
sentMapSize.at(label)+= value;
```

Reply:
Ok